src/backend/utils/adt/orderedsetaggs.c

```diff
@@ -1165,7 +1168,7 @@ mode_final(PG_FUNCTION_ARGS)
 		CHECK_FOR_INTERRUPTS();
 	}
 
-	if (shouldfree && !last_val_is_mode)
+	if (shouldfree && !last_val_is_mode && PointerIsValid(DatumGetPointer(last_val)))
```

Comment:
I'm sorry, but it's 82 chars long.
~~Do we care?~~

Reply:
`pgindent` leave this line unchanged.

Reply:
The restriction in 80 characters in the line [is not so strong](https://www.postgresql.org/docs/9.4/source-format.html#SOURCE-FORMAT). It's up to you to move last conjunct to the next line or leave all the same.

Reply:
I tried several options.

I look that the current version is easier to read
```
	if (shouldfree && !last_val_is_mode && PointerIsValid(DatumGetPointer(last_val)))
		pfree(DatumGetPointer(last_val));
```
than		
```
        if (shouldfree && !last_val_is_mode &&
                PointerIsValid(DatumGetPointer(last_val)))
                pfree(DatumGetPointer(last_val));
```
or
```
	if (shouldfree &&
		!last_val_is_mode &&
		PointerIsValid(DatumGetPointer(last_val)))
		pfree(DatumGetPointer(last_val));

```
or
```
        if (shouldfree
                && !last_val_is_mode
                && PointerIsValid(DatumGetPointer(last_val)))
                pfree(DatumGetPointer(last_val));

```
or even
```
        if (shouldfree
                && !last_val_is_mode
                && PointerIsValid(DatumGetPointer(last_val)))
        {
                pfree(DatumGetPointer(last_val));
        }
```


I think readability of the code is the first priority even if it breaks 80 character line rule as it seems to be OK with [formatting rules](https://www.postgresql.org/docs/9.4/source-format.html#SOURCE-FORMAT) (link from previous comment):

> Limit line lengths so that the code is readable in an 80-column window. (This doesn't mean that you must never go past 80 columns. For instance, breaking a long error message string in arbitrary places just to keep the code within 80 columns is probably not a net gain in readability.)

What is your opinion?

Reply:
For me it's OK in all options except, maybe, latest.
It's all up to you. If you decide to leave it as is, then we'll not need to approve it again after dismissing :smile:

Reply:
If break boolean expression line with `PointerIsValid` indented to the same column as statement `pfree` - I think it makes it harder to read without braces.

The point of review is to find faults and improve code where possible.
If we can pick the best option - I will update the code.
Otherwise it will remain at is it now.

If it is up to me - no need to change.

Reply:
Then I mark it as resolved. Thanks.